coldfront/core/test_helpers/factories.py

```diff
@@ -30,24 +104,86 @@ class Meta:
         model = GrantStatusChoice
 
 
+
+### Project factories ###
+
 class ProjectStatusChoiceFactory(DjangoModelFactory):
+    """Factory for ProjectStatusChoice model"""
     class Meta:
         model = ProjectStatusChoice
+        # ensure that names are unique
+        django_get_or_create = ('name',)
+    # randomly generate names from list of default values
+    name = FuzzyChoice(project_status_choice_names)
 
 
 class ProjectFactory(DjangoModelFactory):
     class Meta:
         model = Project
+        django_get_or_create = ('title',)
 
-    title = 'Test project!'
     pi = SubFactory(UserFactory)
-    description = 'This is a project description.'
+    title = factory.Faker('project_title')
+    description = factory.Faker('sentence')
     field_of_science = SubFactory(FieldOfScienceFactory)
     status = SubFactory(ProjectStatusChoiceFactory)
     force_review = False
     requires_review = False
 
 
+class ProjectUserRoleChoiceFactory(DjangoModelFactory):
+    class Meta:
+        model = ProjectUserRoleChoice
+        django_get_or_create = ('name',)
+    name = 'User'
+
+
+class ProjectUserStatusChoiceFactory(DjangoModelFactory):
+    class Meta:
+        model = ProjectUserStatusChoice
+        django_get_or_create = ('name',)
+    name = 'Active'
```

Comment:
Here the "name" value is hard-coded instead of taken from a list of possible values because it's the value that's most useful during testing. A different approach was used in ProjectStatusChoiceFactory (line 117), where possible choices were randomly assigned.